### PR TITLE
Move stdlib.c to briey/lib, add UART out to memtest

### PR DESCRIPTION
**Pull request by tianyouw**
Move stdlib.c to briey/lib/ so it can be used by other programs. Add UART printing to memtest.

### CODE REVIEW

**Review by Jiangyi — APPROVED** [review]

File: projects/briey/memtest/src/main.c [hunk 1]
```diff
@@ -1,26 +1,39 @@
...
+#define UART_BAUD     (115200U)
 
 extern void flushDataCache(); // From /vga/src/crt.S
 extern int _heap_start, _heap_end;
 volatile uint8_t state = 0;
 
-void fail(){
-    GPIO_A_BASE->OUTPUT = 0xFFFFFFFF ^ 0x04000000 ^ 0x000000FF; // All red LEDs
+void fail(uint32_t errorAddr){
+    GPIO_A_BASE->OUTPUT = (0xFFFFFFFF ^ 0x04000000 ) ^ 0x000000FF; // All red LEDs
```
Comment (Denton-L) [inline projects/briey/memtest/src/main.c]: (0xFFFFFFFF ^ 0x04000000 ) -> ~0x04000000

File: projects/briey/memtest/src/main.c [hunk 2]
```diff
@@ -1,26 +1,39 @@
...
+    printf("Error detected at addr %d\r\n", errorAddr);
```
Comment (Denton-L) [inline projects/briey/memtest/src/main.c:16]: Do we need the \r? (Same for everything below)

File: projects/briey/memtest/src/main.c [hunk 3]
```diff
@@ -1,26 +1,39 @@
...
     GPIO_A_BASE->OUTPUT_ENABLE = 0xFFFFFFFF;
     GPIO_A_BASE->OUTPUT = 0x04000000; // LEDG8
 
+	Uart_Config uartConfig;
+	uartConfig.dataLength = UART_DATA_LEN;
+	uartConfig.parity = NONE;
+	uartConfig.stop = ONE;
+	uartConfig.clockDivider = (CORE_HZ / UART_DATA_LEN / UART_BAUD) - 1;
+	uart_applyConfig(UART,&uartConfig);
+
```
Comment (Denton-L) [inline projects/briey/memtest/src/main.c:36]: Change tabs to spaces

**Review by Denton-L — CHANGES_REQUESTED** [review]

**Review by JYYao — APPROVED** [review]

File: projects/briey/memtest/src/main.c (hunk 2 above)
Comment (tianyouw) [reply projects/briey/memtest/src/main.c:16]: Need it or Putty output will be misaligned: https://superuser.com/questions/654490/putty-new-line-not-working-properly

**Review by Denton-L — APPROVED** [review]

**Pull request merged**